app/controllers/sales_logs_controller.rb

```diff
@@ -61,6 +64,17 @@ def permitted_log_params
 
 private
 
+  def extract_bulk_upload_from_session_filters
+    filter_service = FilterService.new(current_user:, session:)
+    @bulk_upload = filter_service.bulk_upload
+  end
+
+  def redirect_if_bulk_upload_resolved
+    if @bulk_upload && @bulk_upload.sales? && @bulk_upload.sales_logs.in_progress.count.zero?
```

Comment:
We could replace the first two conditions with `@bulk_upload&.sales?`

Reply:
or even just replace the whole line with `@bulk_upload&.sales_logs.in_progress.count.zero?`  I think

Reply:
i think we can change to the first one. second one i think will have a bug cause of the nature that bulk upload can be lettings or sales and we have yet to do work to split/scope filters out for different log types